menas/src/main/scala/za/co/absa/enceladus/menas/controllers/VersionedModelController.scala

```diff
@@ -158,4 +158,17 @@ abstract class VersionedModelController[C <: VersionedModel with Product with Au
     versionedModelService.disableVersion(name, v)
   }
 
+  @PutMapping(Array("/lock/{name}"))
+  @ResponseStatus(HttpStatus.OK)
+  def lock(@PathVariable name: String): CompletableFuture[UpdateResult] = {
+    versionedModelService.setLock(name, isLocked = true)
```

Comment:
Why the `setLock` method has to figure out the authenticated user itself and it's not coming from API itself (`@AuthenticationPrincipal principal: UserDetails`), like in most other places (well except `delete`)?